appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/impl/MetricsConfigParserUtil.java

```diff
@@ -0,0 +1,175 @@
+            String metricName = resultKeyValueSplit[0];
+            if(resultKeyValueSplit.length == 1) {
+                //evaluate when no value, disabled
+            } else {
+                Duration[] bucketValues = Arrays.asList(resultKeyValueSplit[1].split(PROPERTY_VALUE_SEPARATOR)).stream()
+                        .map(MetricsConfigParserUtil::evaluateTimerBucketValue)
+                        .filter(d -> d != null).toArray(Duration[]::new);
+                customBucket = new TimerMetricsBucket(metricName, 
+                        Arrays.stream(bucketValues).mapToDouble(Duration::toNanos).boxed().toArray(Double[]::new));
+            }
+            metricTimerCollection.addFirst(customBucket);
+        }
+        return metricTimerCollection;
+    }
+
+    public static synchronized Collection<MetricsCustomPercentile> processPercentileMap(String appName) {
+        Optional<String> customPercentiles = ConfigProvider.getConfig().getOptionalValue(METRIC_PERCENTILES_PROPERTY, String.class);
+        return (customPercentiles.isPresent()) ? MetricsConfigParserUtil.parsePercentile(customPercentiles.get()) : null;
+    }
+    
+    public static Double evaluatePercentileValue(String percentile) {
+        if(percentile.matches("[0][.][0-9]+")) {
+            return Double.parseDouble(percentile);
+        } else {
+            logger.info(String.format("Error when trying to read property %s with %s name", METRIC_PERCENTILES_PROPERTY, percentile));
+            return null;
+        }
+    }
+
+    public static Double evaluateHistogramBucketValue(String bucket) {
+        if(bucket.matches("[0-9]+[.]*[0-9]*")) {
+            return Double.parseDouble(bucket);
+        } else {
+            logger.info(String.format("Error when trying to read property %s with %s name", METRIC_HISTOGRAM_BUCKETS_PROPERTY, bucket));
+            return null;
+        }
+    }
+
+    public static Duration evaluateTimerBucketValue(String bucket) {
+        bucket = bucket.trim();
+        if(bucket.matches("[0-9]+ms")) { //case ms
+            return Duration.ofMillis(Long.parseLong(bucket.substring(0, bucket.length() - 2)));
+        } else if(bucket.matches("[0-9]+s")) { //case s
+            return Duration.ofSeconds(Long.parseLong(bucket.substring(0, bucket.length() - 1)));
+        } else if(bucket.matches("[0-9]+m")) { //case m
+            return Duration.ofMinutes(Long.parseLong(bucket.substring(0, bucket.length() - 1)));
+        } else if(bucket.matches("[0-9]+h")) { //case h
+            return Duration.ofHours(Long.parseLong(bucket.substring(0, bucket.length() - 1)));
+        } else if(bucket.matches("[0-9]+")) { //normal
+            return Duration.ofMillis(Long.parseLong(bucket));
+        } else {
+            logger.info(String.format("Error when trying to read property %s with %s name", METRIC_TIMER_BUCKETS_PROPERTY, bucket));
+            return null;
+        }
```

Comment:
I think a switch would be easier to read here

Reply:
yeah but the needed to evaluate the regex with the matches call is adding complexity to evaluate the expression. I reviewed the switch pattern match and can help to improve this switch but that will couple the code with JDK 17 as minimum to have that functionality available